pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -339,6 +351,21 @@ public String getTaskType() {
           .info("Finished generating task configs for table: {} for task: {}, numTasks: {}", offlineTableName, taskType,
               pinotTaskConfigsForTable.size());
     }
+
+    // Reset watermarks for invalid tables. This covers the metrics clean up when the table is removed or the merge
+    // config is added and then removed.
+    LeadControllerManager leadControllerManager = _clusterInfoAccessor.getLeaderControllerManager();
+    List<String> cleanUpCandidates = new ArrayList<>();
+    for (String tableNameWithType : _mergeRollupWatermarks.keySet()) {
+      if (!candidateMergeTables.contains(tableNameWithType) || !leadControllerManager
```

Comment:
(Critical) This will remove all the metrics when someone triggers an adhoc task which only generate task for one table

Reply:
Added the `TODO` comment in `cleanUpMetrics()`. I will address this from the separate PR.